sdk/keyvault/Azure.Security.KeyVault.Administration/src/RoleAssignmentScope.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Security.KeyVault.Administration
+{
+    /// <summary>
+    /// A scope of the role assignment.
+    /// </summary>
+    public readonly struct RoleAssignmentScope : IEquatable<RoleAssignmentScope>
+    {
+        internal const string RootValue = "/";
+        internal const string KeysValue = "/keys";
+
+        private readonly string _value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoleAssignmentScope"/> structure.
+        /// </summary>
+        /// <param name="value">The string value of the instance.</param>
+        public RoleAssignmentScope(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoleAssignmentScope"/> structure.
+        /// </summary>
+        /// <param name="ResourceId">The Resource Id for the given Resource</param>
+        public RoleAssignmentScope(Uri ResourceId)
+        {
+            _value = ResourceId.AbsolutePath ?? throw new ArgumentNullException(nameof(ResourceId));
+        }
```

Comment:
This is the middle ground approach I settled on for allowing developers to "bring their own resource ID" and represent it properly as a `RoleAssignmentScope`. 
ex.
```csharp
// Some Key fetched from the Keys client.
KeyVaultKey myKey;

// Get the Key specific resources scope.
RoleAssignmentScope keySpecificScope = new RoleAssignmentScope(myKey.Id);

// Use the scope to assign a role assignment.
RoleAssignment assignment = await Client.CreateRoleAssignmentAsync(keySpecificScope , properties, roleAssignmentId);
```

Reply:
I would move the roleAssignmentId into an optional parameter or Options-type class. We should generate it by default. I doubt very few people would need to code it outside of tests.

Reply:
It should have been left out of this example, but it is an optional parameter.